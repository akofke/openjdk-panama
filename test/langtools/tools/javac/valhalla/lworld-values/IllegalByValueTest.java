/*
 * @test /nodynamiccopyright/
 * @bug 8209400
 * @summary Allow anonymous classes to be value types
 * @compile/fail/ref=IllegalByValueTest.out -XDrawDiagnostics -XDdev IllegalByValueTest.java
 */

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;
import java.util.Comparator;

public class IllegalByValueTest {

    @Target(ElementType.TYPE_USE)
    @interface Annot {
    }

    public static void main(String[] args) {
        // Error cases.
        __MakeDefault __ByValue IllegalByValueTest() {};
        __MakeDefault __ByValue __ByValue IllegalByValueTest() {};
        __MakeDefault @Annot __ByValue IllegalByValueTest() {};
        new __ByValue @Annot __ByValue Comparable <String>() {};
        int [] ia = new __ByValue int[10];
        new __ByValue String("Hello");
    }
}
