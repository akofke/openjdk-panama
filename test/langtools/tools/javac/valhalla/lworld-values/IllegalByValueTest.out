IllegalByValueTest.java:20:23: compiler.err.mod.not.allowed.here: value
IllegalByValueTest.java:21:23: compiler.err.mod.not.allowed.here: value
IllegalByValueTest.java:21:33: compiler.err.mod.not.allowed.here: value
IllegalByValueTest.java:22:30: compiler.err.mod.not.allowed.here: value
IllegalByValueTest.java:23:30: compiler.err.mod.not.allowed.here: value
IllegalByValueTest.java:24:25: compiler.err.mod.not.allowed.here: value
IllegalByValueTest.java:25:13: compiler.err.mod.not.allowed.here: value
7 errors
